include/posix/pthread.h

```diff
@@ -30,6 +31,9 @@ enum pthread_state {
 struct posix_thread {
 	struct k_thread thread;
 
+	/* List of keys that thread has called pthread_setspecific() on */
+	sys_dlist_t key_list;
```

Comment:
I don't see why this has to be a dlist.  You don't ever do anything but search the list linearly from front to back, and only remove elements having found them via linear search.  This should be doable with an slist, no?

Reply:
Ah, good catch. I'm not sure why I used dlist in the beginning. I see the singly linked list implementation is circular too. You are right, I will make use of `slist` only to save space.